diff --git a/src/aarch64/ffi.c b/src/aarch64/ffi.c
index 2299ce4..0f99673 100644
--- a/src/aarch64/ffi.c
+++ b/src/aarch64/ffi.c
@@ -80,10 +80,10 @@ ffi_clear_cache (void *start, void *end)
 {
 #if defined (__clang__) && defined (__APPLE__)
   sys_icache_invalidate (start, (char *)end - (char *)start);
-#elif defined (__GNUC__)
-  __builtin___clear_cache (start, end);
 #elif defined (_WIN32)
   FlushInstructionCache(GetCurrentProcess(), start, (char*)end - (char*)start);
+#elif defined (__GNUC__)
+  __builtin___clear_cache (start, end);
 #else
 #error "Missing builtin to flush instruction cache"
 #endif
